mlflow/models/cli.py

```diff
@@ -155,6 +155,35 @@ def prepare_env(
     ).prepare_env(model_uri=model_uri)
 
 
+@commands.command("generate-dockerfile")
+@cli_args.MODEL_URI_BUILD_DOCKER
+@click.option("--output-directory",
+              "-d",
+              default="generate-dockerfile-output",
+              help="output directory for generating dockerfile")
+@cli_args.ENV_MANAGER
+@cli_args.MLFLOW_HOME
+@cli_args.INSTALL_MLFLOW
+@cli_args.ENABLE_MLSERVER
+def generate_dockerfile(model_uri, output_directory, env_manager, mlflow_home, install_mlflow, enable_mlserver):
```

Comment:
Can we add a test for this command? I think that https://github.com/mlflow/mlflow/blob/master/tests/models/test_cli.py is a good suite to add to

Reply:
I will add test once we can find consensus on directory parameter in function, since it can potentially change a function hence a test.
At the moment I'm just not sure if we might need to change a function to accept a path to Dockerfile instead of directory :sweat_smile: , so until you approve decision to leave directory as an argument or decide to change a function's aruments, I will hold on a test for a little bit :sweat_smile:

Reply:
Responded here: https://github.com/mlflow/mlflow/pull/6591/files#r956774884. I think that a directory output makes sense :)

Reply:
Yep, @dbczumar, sorry for the delay --- we had country holidays up until Wednesday :)
I'm gonna try and update pull request today to address all remaining comments and issues

Reply:
I have written a test to ensure that `model_dir` and `Dockerfile` are generated and `Dockerfile` is not empty
However, I am somewhat struggling with test launching --- I am running `dev/run-python-tests` but it fails in 3 cases for unrelated reasons (not installed `kubernetes`, `google.cloud` and `pyarrow` modules)
When I tried to purposefully fail tests they are also didn't appear as failed once I ran the script.

Could it be related to issue https://github.com/mlflow/mlflow/issues/1096?

Reply:
@harupy @dbczumar  Should we add `kubernetes`, `google-cloud-storage` and `pyarrow` to `requirements/test_requirements.txt` to ensure valid launch of `dev/run-python-tests`?
Should it be a separate issue?